File: lib/src/data/models/product_variant.dart
class ProductVariant {
  final int variantId;      // PK
  final int productId;      // FK
  final String variantType; // size, color...
  final String variantValue;// "37", "Red"...
  final double price;
  final int stock;

  ProductVariant({
    required this.variantId,
    required this.productId,
    required this.variantType,
    required this.variantValue,
    required this.price,
    required this.stock,
  });
}

File: src/main/java/com/example/sneakerx/entities/ProductVariant.java
package com.example.sneakerx.entities;

import com.example.sneakerx.entities.enums.VariantType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "product_variants")
@Builder
public class ProductVariant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer variantId;

    @Enumerated(EnumType.STRING)
    private VariantType variantType;

    private String variantValue;
    private Double price;
    private Integer stock;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Product product;
}
